src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private final double side;
+
+    public Square(double side) {
+        this.side = side;
+        setColor(new ColorSupplier().getRandomColor());
```

Comment:
Don't pass expressions in the constructor. Instead, create a variable for the color and pass it to the constructor for better code readability. Also, consider generating random values in `FigureSupplier` rather than in the `Square` class.